### ISSUE TITLE
Sample Images During Training to Check Generation Quality

### ISSUE THREAD

**Issue by EugenHotaj** [EugenHotaj/pytorch-generative#17]
Currently, the only way to measure model quality during training is by monitoring the evaluation loss. However, a low loss does not always correspond to high quality generation (e.g. see https://arxiv.org/abs/1511.01844). Furthermore, for autoregressive models, if the model is not correctly masked, it will get very low loss by peeking into the future, but the generation quality will be extremely bad. 

In #16, one suggestion was to periodically sample from the model as it trains (e.g. every 10 epochs) and display these samples using TensorBoard. We should be able to easily support this in the [Trainer](https://github.com/EugenHotaj/pytorch-generative/blob/master/pytorch_generative/trainer.py#L9).
